omniscidb/QueryEngine/CostModel/DataSources/DwarfBench.cpp

```diff
@@ -18,147 +18,76 @@
 
 namespace costmodel {
 
-DwarfBench::DwarfBench()
-    : DataSource(DataSourceConfig{
-          .dataSourceName = "DwarfBench",
-          .supportedDevices = {ExecutorDeviceType::CPU, ExecutorDeviceType::GPU},
-          .supportedTemplates = {AnalyticalTemplate::GroupBy,
-                                 AnalyticalTemplate::Join,
-                                 AnalyticalTemplate::Reduce,
-                                 AnalyticalTemplate::Scan}}) {}
-
-const std::string DwarfBench::sizeHeader = "buf_size_bytes";
-const std::string DwarfBench::timeHeader = "total_time";
-
-std::string DwarfBench::getDwarfBenchPath() {
-  static const char* DWARF_BENCH_PATH = std::getenv("DWARF_BENCH_PATH");
-
-  if (DWARF_BENCH_PATH == NULL) {
-    throw DwarfBenchException("DWARF_BENCH_PATH environment variable not set");
-  } else {
-    return DWARF_BENCH_PATH;
-  }
-}
-
-Detail::DeviceMeasurements DwarfBench::getMeasurements(
+DwarfBenchDataSource::DwarfBenchDataSource()
+    : DataSource(DataSourceConfig{.dataSourceName = "DwarfBench",
+                                  .supportedDevices = {ExecutorDeviceType::CPU},
+                                  .supportedTemplates = {AnalyticalTemplate::GroupBy,
+                                                         AnalyticalTemplate::Join,
+                                                         AnalyticalTemplate::Scan}}) {}
+
+Detail::DeviceMeasurements DwarfBenchDataSource::getMeasurements(
     const std::vector<ExecutorDeviceType>& devices,
     const std::vector<AnalyticalTemplate>& templates) {
   Detail::DeviceMeasurements dm;
-  boost::filesystem::path dwarf_path = getDwarfBenchPath();
-
-  if (!boost::filesystem::exists(dwarf_path / "results")) {
-    boost::filesystem::create_directory(dwarf_path / "results");
-  }
-
   for (AnalyticalTemplate templ : templates) {
+    if (!isTemplateSupported(templ)) continue;    // Or should we throw an exception?
```

Comment:
Since this will be a part of the calibration this should be an implementation detail for the query engine. You can throw an exception if we handle it in the calibration code, but I don't see much reason to. Trying to execute an unsupported template looks like incorrect application logic to me. How about a `CHECK` instead?